biogeochem/EDPatchDynamicsMod.F90

```diff
@@ -98,11 +102,9 @@ module EDPatchDynamicsMod
   use SFParamsMod,            only : SF_VAL_CWD_FRAC
   use EDParamsMod,            only : logging_event_code
   use EDParamsMod,            only : logging_export_frac
+  use EDParamsMod,            only : maxpatches_by_landuse
   use FatesRunningMeanMod,    only : ema_sdlng_mdd
   use FatesRunningMeanMod,    only : ema_sdlng_emerg_h2o, ema_sdlng_mort_par, ema_sdlng2sap_par
-  use EDParamsMod,            only : maxpatch_primary
-  use EDParamsMod,            only : maxpatch_secondary
-  use EDParamsMod,            only : maxpatch_total
   use FatesRunningMeanMod,    only : ema_24hr, fixed_24hr, ema_lpa, ema_longterm
   
```

Comment:
This routine in general, is getting out of hand regards to comments and transparency.  We have blocks of code that have no comments explaining what is about to be performed.  This may had not been a big deal in the past when it performed a small number of features, but its long enough now that we need more explanations.

Reply:
@rgknox I added a bunch of comments to the spawn_patches subroutine to try to better orient the reader on what is going on there and why.